src/components/10-atoms/input-text/index.js

```diff
@@ -248,6 +252,31 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
     this.nativeInput = this.querySelector('input');
   }
 
+  _formatCurrency(value) {
+    const { currency, type } = this;
+
+    if (currency && type === 'text' && !this.currencyFormatter) {
+      // just create a new Intl if it does not exist
+      this.currencyFormatter = new Intl.NumberFormat('de-CH', {
+        style: 'currency',
+        currency,
+      });
+    }
+
+    if (this.currencyFormatter) {
```

Comment:
i don't really like the structure of those if-else blocks in here.

line 266 checks if currencyformatter is set, which is an obsolete check if the type is "number" (which is already checked previously). this is not a clean approach.

better would be (pseudocode):

```
_formatCurrency(value) {
  if (currency && type=="number") { // if not type currency or number, don't look any further...
    if (!this.currencyFormatter) {
       // INIT
    }
    // proceed the formatting...
  }
  return value
}
`
``

Reply:
done